File: Dataset/Leetcode/train/125/524.cpp
 class Solution {
public:
    bool XXX(string s) {
        if (s == "") return true;

        string str = "";
        for (int i = 0; i < s.size(); i++){
            if (isdigit(s[i])) str += s[i];
            if (isalpha(s[i])) str += tolower(s[i]);
        }

        for (int i = 0, j = str.size() -1; i <= j; i++, j--){
            if (str[i] != str[j]) return false;
        }

        return true;

    }
};

File: Dataset/Leetcode/train/98/524.java
 class Solution {
    ArrayList<Integer>list=new ArrayList();
    public boolean XXX(TreeNode root) {
        if(root==null) return true;
        //中序遍历，形成有序链表
        return midOrder(root);
    }
    public boolean midOrder(TreeNode root){
        Stack<TreeNode>stack=new Stack();
        TreeNode p=root;
        while(p!=null||!stack.empty()){
            if(p!=null){
                stack.push(p);//先缓存p,再向左访问
                p=p.left;
            }
            else{
                p=stack.pop();
                int size=list.size();
                if(size==0||list.get(size-1)<p.val)//在中序遍历的过程中如果发现非递增，则直接返回false
                   list.add(p.val);
                else return false;
                p=p.right;
            }
        }
        return true;
    }
}

